ruby/ql/lib/codeql/ruby/frameworks/ActionCable.qll

```diff
@@ -26,4 +28,33 @@ module ActionCable {
       }
     }
   }
+
+  /**
+   * The data argument in an RPC endpoint method on a subclass of
+   * `ActionCable::Channel::Base`, considered as a remote flow source.
+   */
+  class ActionCableChannelRpcParam extends RemoteFlowSource::Range {
+    ActionCableChannelRpcParam() {
+      exists(DataFlow::MethodNode m |
+        // Any method on a subclass of `ActionCable::Channel::Base`
+        // automatically becomes an RPC endpoint
+        m =
+          DataFlow::getConstant("ActionCable")
+              .getConstant("Channel")
+              .getConstant("Base")
+              .getADescendentModule()
+              .getAnOwnInstanceMethod() and
+        // as long as it's public
+        m.asCallableAstNode().isPublic() and
```

Comment:
Would you mind adding `isVisible()` to `DataFlow::MethodNode`?

From JS my experience was that we should have added a lot of these trivial things sooner rather than later.

Reply:
Sure, but wouldn't it be better to add `isPublic`, `isPrivate`, and `isProtected`?

Reply:
Yes, you're right, I don't know where I got `isVisible()` from 🤦